models/DataObject/ClassDefinition/Data/Relations/AllowAssetRelationTrait.php

```diff
@@ -16,6 +16,10 @@ trait AllowAssetRelationTrait
      */
     protected function allowAssetRelation($asset)
     {
+        if ($asset instanceof Asset && $asset->getId() <= 0) {
```

Comment:
instanceof check really needed? as already done in checkValidity

Reply:
I would keep it for consistency as the same is not true for objects. Besides that, this is not an internal method.

Reply:
then better would be `!$asset instanceof Asset || $asset->getId() <= 0`

Reply:
done